pkg/store/bucket.go

```diff
@@ -1161,6 +1188,23 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 		s.metrics.seriesGetAllDuration.Observe(stats.GetAllDuration.Seconds())
 		s.metrics.seriesBlocksQueried.Observe(float64(stats.blocksQueried))
 	}
+
+	// Send response hints at the beginning of the stream. This helps the querier know very early in the
```

Comment:
This needs to be double checked with Mimir and Cortex - cc @pracucci and @yeya24 

We don't break API but maybe they were relying on those send at the end?